rdr_service/tools/tool_libs/thebasics_analyzer.py

```diff
@@ -0,0 +1,522 @@
+        #  If the prior_response is None (first payload), all content will be displayed as a new (+) answer
+
+        diff_details = list()
+        prior_response_keys = prior_response.keys() if prior_response else []
+        curr_response_keys = curr_response.keys()
+        key_set = set().union(prior_response_keys, curr_response_keys)
+
+        for key in sorted(key_set):
+            if key in prior_response_keys and key not in curr_response_keys:
+                diff_details.append(('-', key))
+            else:
+                answer = curr_response.get(key)
+                # Redact answers to free text fields (PII) unless they were skipped, or redaction is disabled
+                if answer.lower() != 'pmi_skip' and key in REDACTED_FIELDS and not self.args.no_redact:
+                    answer = '<redacted>'
+                if key in curr_response_keys and key not in prior_response_keys:
+                    diff_details.append(('+', key, answer))
+                elif prior_response[key] != curr_response[key]:
+                    diff_details.append(('!', key, answer))
+                else:
+                    diff_details.append(('=', key))
+
+        return diff_details
+
+    def get_response_as_dict(self, response_id : int, session=None) -> dict:
+        """
+        Generate a dict of a TheBasics response.  Includes meta data keys/values and an answers nested dict
+        with question code keys and answer values.  The answer value is from a COALESCE of the
+        QuestionnaireResponseAnswer table's valueCodeId, valueString, valueBoolean, valueInteger, valueDate,
+        and valueDateTime columns
+        :param response_id:   Integer questionnaire response id
+        :param session:       A DAO session() object, if one has already been instantiated.
+        """
+        response_dict = {'answers': dict()}
+        if not session:
+            close_session = True
+            session = self.ro_dao.session()
+        else:
+            close_session = False
+
+        answer = aliased(Code)
+        # One row returned for each questionnaire_response_answer entry associated with this questionnaire_response_id
+        # The meta data fields from QuestionnaireResponse remain the same across all rows
+        answer_list = session.query(
+            QuestionnaireResponse.questionnaireResponseId,
+            QuestionnaireResponse.answerHash,
+            QuestionnaireResponse.authored,
+            QuestionnaireResponse.externalId,
+            Code.value.label('question_code_value'),
+            func.coalesce(answer.value,
+                          QuestionnaireResponseAnswer.valueString,
+                          QuestionnaireResponseAnswer.valueBoolean,
+                          QuestionnaireResponseAnswer.valueInteger,
+                          QuestionnaireResponseAnswer.valueDate,
+                          QuestionnaireResponseAnswer.valueDateTime
```

Comment:
Should this consider valueDecimal as well?

Reply:
Updated in latest commit.  Also brought in `valueSystem` and `valueUri` for completeness